### PR TITLE
Allow running with different game IDs for debug builds of osu!lazer

### PR DESCRIPTION
**Pull request by frenzibyte**
Making things easy to test development server with multiple game instances running simultaneously, without having to log out and in between accounts in each instance, or manually modifying the game name and potentially causing a `BadImageFormatException` in the already running instance.

Of course, this argument should never be accessible by normal users so I've limited access to `DebugUtils.IsDebugBuild` only.

### CODE REVIEW

File: osu.Desktop/Program.cs [hunk 1]
```diff
@@ -48,16 +57,10 @@ public static int Main(string[] args)
                         return 0;
                 }
 
-                switch (args.FirstOrDefault() ?? string.Empty)
```
Comment (peppy) [inline osu.Desktop/Program.cs:51]: Why is this changed from a `switch`?

Comment (frenzibyte) [reply osu.Desktop/Program.cs:51]: Because there can now be more than one argument, I've changed this to use `args.Contains("--tournament")` rather than assuming the first argument will always have that.

File: osu.Desktop/Program.cs [hunk 2]
```diff
@@ -23,7 +23,16 @@ public static int Main(string[] args)
             // Back up the cwd before DesktopGameHost changes it
             var cwd = Environment.CurrentDirectory;
 
-            using (DesktopGameHost host = Host.GetSuitableHost(@"osu", true))
+            string gameName = @"osu";
+
+            if (DebugUtils.IsDebugBuild)
+            {
+                var customNameArg = args.SingleOrDefault(s => s.StartsWith(@"--name=", StringComparison.Ordinal));
```
Comment (peppy) [inline osu.Desktop/Program.cs:30]: `--name` is a really bad (read: nondescript) choice.

Not really sure if we should be changing the complete game name here, either. Is there a reason you didn't focus directly on the storage rather than the whole game?

File: osu.Desktop/Program.cs (hunk 1 above)
Comment (peppy) [reply osu.Desktop/Program.cs:51]: That's not how it should work. At *very* least a naive approach would split and process the `args`, handling via a `switch` once at a time.

File: osu.Desktop/Program.cs (hunk 2 above)
Comment (frenzibyte) [reply osu.Desktop/Program.cs:30]: Yeah the name of the argument was very arbitrary, maybe `--game-identifier` or something? not sure.

I made this argument change the complete game name as I think it's the best choice to define a whole different place and data for a game instance that can be run side-by-side with the main one.

It seems to be potentially possible to modify `OsuStorage` for that, but I'm not really sure if it's worth adding further logic in the main game codebase for "debug builds"-specific features, rather than going with this simple way of changing the game name.

Unless you think there could potentially be an issue with changing the game name?

File: osu.Desktop/Program.cs (hunk 1 above)
Comment (frenzibyte) [reply osu.Desktop/Program.cs:51]: I don't quite understand what you mean?

File: osu.Desktop/Program.cs (hunk 2 above)
Comment (peppy) [reply osu.Desktop/Program.cs:30]: The only way I'd be happy with this happening is if you make it into an `int` only suffix:

`--debug-client-id=1` - starts osu! with identifier `osu-1`

File: osu.Desktop/Program.cs (hunk 1 above)
Comment (peppy) [reply osu.Desktop/Program.cs:51]:
```csharp
foreach (string arg in args)
{
    switch(arg)
    {
        // handle each arg.
    }
}
```

**Comment by peppy** [comment]
I've added rider configurations to benefit from this new argument. Can you please test to make sure they show up and are usable for you?

**Comment by frenzibyte** [comment]
Shows up and works as expected.

**Comment by peppy** [comment]
@smoogipoo just confirming you're fine with the added project run configuration for this?

**Review by peppy — APPROVED** [review]

**Comment by smoogipoo** [comment]
Is there no better way that this can be done? Having 2+ configurations sounds like a pain, honestly.

I have one `osu! (Parallel)` configuration locally which I used to run multiple instances (however many I want) in parallel by clicking one button. Can it not count the number of processes running or something instead?

**Comment by peppy** [comment]
If you need more than two (I'd argue that's an edge case) you could just `dotnet run` from CLI, no?


https://user-images.githubusercontent.com/191335/126441551-4961fd6b-dcc7-42ef-8d08-b0035c7ddf95.mp4

This is how it looks, for reference.

**Comment by smoogipoo** [comment]
Go with it for now then.

**Comment by peppy** [comment]
While I think automatically doing this on subsequent clients is pretty nice for this specific scenario, it could also lead to potential misfires if a previous instance is still shutting down, so I do want to make this an explicit argument for now.
